### PR TITLE
Markdown datatype support.

### PR DESCRIPTION
**Pull request by jmchilton**
With test tools demonstrating HTML and Markdown outputs from tools.

Implements #3396 as part of the Qiime hackathon.

### CODE REVIEW

File: test/functional/tools/creates_html.xml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+<tool id="creates_html" name="creates_html" version="1.0.0">
+  <command>
+    mkdir '$output.extra_files_path';
```
Comment (nsoranzo) [inline test/functional/tools/creates_html.xml:3]: && should be used to concatenate commands, which also requires the use of CDATA.

File: test/functional/tools/creates_markdown.xml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+<tool id="creates_markdown" name="creates_markdown" version="1.0.0">
+  <command>
+    mkdir '$output.extra_files_path';
```
Comment (nsoranzo) [inline test/functional/tools/creates_markdown.xml:3]: Same here.

File: test/functional/tools/creates_html.xml (hunk 1 above)
Comment (jmchilton) [reply test/functional/tools/creates_html.xml:3]: I don't consider this to be best practice as of 16.04 - if tools specify a profile version of greater than ``16.01`` the commands run in an environment with ``set -e``. I've specified a tool version of 16.04 in a rebase to enable this. Can we call this point resolved or do you still insist on the ``CDATA``.

Comment (nsoranzo) [reply test/functional/tools/creates_html.xml:3]: Works for me in this case.

On the other hand, I think IUC recommendations will stick with && and CDATA for some time for backward compatibility. Also CDATA is always needed for '>'.

**Review by nsoranzo — APPROVED** [review]

**Review by bgruening — APPROVED** [review]

**Comment by bgruening** [comment]
Leaving the merge to Eric!
Thanks John!

**Pull request merged**